src/zc/lockfile/__init__.py

```diff
@@ -27,10 +27,49 @@ class LockError(Exception):
     try:
         import msvcrt
     except ImportError:
-        def _lock_file(file):
-            raise TypeError('No file-locking support on this platform')
-        def _unlock_file(file):
-            raise TypeError('No file-locking support on this platform')
+        try:
+            import java.lang.Exception
+            import threading
+            import os.path
+        except ImportError:
+            def _lock_file(file):
+                raise TypeError('No file-locking support on this platform')
+            def _unlock_file(file):
+                raise TypeError('No file-locking support on this platform')
+        else:
+            # Jython Java file locks are per-VM, so they can't be used
```

Comment:
It took me several reads of this comment and the related code to get a grasp of what's happening. Let me see if I understand correctly - Java file locks only provide for synchronization across processes (VM instances), but for different threads within a process, any number of threads will acquire the same lock. Perhaps you could include text like that in the comment? Also, an authoritative reference to a description of this behavior would be nice (if available).

Reply:
Yes, that's what it means (and I thought what it said 😄 ). The authoritative link would be to the [tryLock javadoc](https://docs.oracle.com/javase/10/docs/api/java/nio/channels/FileChannel.html#tryLock(long,long,boolean)): 

> File locks are held on behalf of the entire Java virtual machine. They are not suitable for controlling access to a file by multiple threads within the same virtual machine.